macros/calendar_date/string_to_date.sql

```diff
@@ -0,0 +1,152 @@
+{%- macro string_to_date(date_string, format='yyyy-mm-dd') -%}
+    {{ adapter.dispatch('string_to_date', 'dbt_date') (date_string, format) }}
+{%- endmacro -%}
+
+{%- macro default__string_to_date(date_string, format) -%}
+-- redshift and postgres are actually pretty good at detecting date strings. we'll add compatibility for Day/Month/Year format
+
+{% if format|lower in ('d/m/yy', 'dd/mm/yy') %}
```

Comment:
If we're always evaluating the lower-case version of `format` we should only call `lower` once at the beginning to keep it DRY.